kraken-app/kraken-app-portal/src/pages/NewAPIServer/components/SelectAPIServer.tsx

```diff
@@ -4,11 +4,17 @@ import { Text } from "@/components/Text";
 import { useGetValidateServerName } from '@/hooks/product';
 import { useAppStore } from '@/stores/app.store';
 import { validateServerName, validateURL } from '@/utils/helpers/validators';
-import { Form, Input } from "antd";
+import { Form, FormInstance, Input } from "antd";
+import { useMemo } from 'react';
 
-const SelectAPIServer = () => {
+type Props = {
+  form?: FormInstance<any>;
+};
+
+const SelectAPIServer = ({ form }: Props) => {
   const { currentProduct } = useAppStore();
   const { mutateAsync: validateName } = useGetValidateServerName();
+  const originalName = useMemo(() => (form?.getFieldsValue(["name"]).name ?? null), [form]);
```

Comment:
Hi Bartek, form?.getFieldsValue(["name"]) can possibily be undefined
```suggestion
  const originalName = useMemo(() => (form?.getFieldsValue(["name"])?.name ?? null), [form]);
```